mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -1770,3 +1931,98 @@ def _get_search_experiments_order_by_clauses(order_by):
         order_by_clauses.append((SqlExperiment.experiment_id, False))
 
     return [col.asc() if ascending else col.desc() for col, ascending in order_by_clauses]
+
+
+def _get_orderby_clauses_for_search_traces(order_by_list: List[str], session):
+    """Sorts a set of runs based on their natural ordering and an overriding set of order_bys.
+    Runs are naturally ordered first by start time descending, then by run id for tie-breaking.
+    """
+    clauses = []
+    ordering_joins = []
+    observed_order_by_clauses = set()
+    select_clauses = []
+
+    for clause_id, order_by_clause in enumerate(order_by_list):
+        (key_type, key, ascending) = SearchTraceUtils.parse_order_by_for_search_traces(
+            order_by_clause
+        )
+
+        if SearchTraceUtils.is_attribute(key_type, key, "="):
+            order_value = getattr(SqlTraceInfo, key)
+        else:
+            if SearchTraceUtils.is_tag(key_type, "="):
+                entity = SqlTraceTag
+            elif SearchTraceUtils.is_request_metadata(key_type, "="):
+                entity = SqlTraceRequestMetadata
+            else:
+                raise MlflowException(
+                    f"Invalid identifier type '{key_type}'",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+            # Tags and request metadata requires a join to the main table (trace_info)
+            subquery = session.query(entity).filter(entity.key == key).subquery()
+            ordering_joins.append(subquery)
+            order_value = subquery.c.value
+
+        case = sql.case((order_value.is_(None), 1), else_=0).label(f"clause_{clause_id}")
+        clauses.append(case.name)
+        select_clauses.append(case)
+        select_clauses.append(order_value)
+
+        if (key_type, key) in observed_order_by_clauses:
+            raise MlflowException(f"`order_by` contains duplicate fields: {order_by_list}")
+        observed_order_by_clauses.add((key_type, key))
+        clauses.append(order_value if ascending else order_value.desc())
+
+    # Add descending trace start time as default ordering and a tie-breaker
+    if (
+        SearchTraceUtils._ATTRIBUTE_IDENTIFIER,
+        SqlTraceInfo.timestamp_ms.key,
+    ) not in observed_order_by_clauses:
+        clauses.append(SqlTraceInfo.timestamp_ms.desc())
```

Comment:
In order to provide a tie-breaker for pagination clean cut-off purposes, we might want to insert a natural order by clause of the request_id to ensure that within a given timestamp (no matter how unlikely it seems that this might be), we always have a unique and clean break point for the pagination logic.

Reply:
Great idea! The millisecond timestamp can be easily conflict especially during parallel execution. Will add `request_id` as the final tie-breaker.